### ISSUE TITLE
[BUG/ISSUE] AOD does not get archived in ND51/ND51b satellite timeseries diagnostic

### ISSUE THREAD

**Issue by yantosca** [geoschem/geos-chem#260]
I am submitting an issue on behalf of Shixian Zhai (zhaisx [at] g.harvard.edu)
 
>I am trying to output AOD from GEOS-Chem version 12.7.1 using ND51 and ND51b.
I compile GEOS-Chem with the BPCH_DIAG=y option. However, in the ts_satellite.YYYYMMDD.bpch file I get, there are only species concentration output but no AOD output.
 
>My input.geos file on ND51 and ND51b is as follows:
```
%%% ND51 MENU %%%       :
Turn on ND51 diagnostic : T
LT avg timeseries file  : ts_satellite.YYYYMMDD.bpch
Output as HDF5?         : F
Tracers to include      : 1 2 26 27 31 32 34 35 36 37 38 39 42 64 74 505 507 517 518 519 520 521 522 523 524 525 526 527 528 529
GMT Hour for disk write :   0
Avg Period [LT hours]   :   9.5  16.5
IMIN, IMAX of region    :   56  67
JMIN, JMAX of region    :   27  37
LMIN, LMAX of region    :   1   47
-------------------------------------------------------------------------------
%%% ND51b MENU %%%      :
Turn on ND51b diagnstic : T
LT avg timeseries file  : ts_13_15.NA.YYYYMMDD.h5
Output as HDF5?         : F
Tracers to include      : 1 2 26 27 31 32 34 35 36 37 38 39 42 64 74 505 507 517 518 519 520 521 522 523 524 525 526 527 528 529
GMT Hour for disk write :  0
Avg Period [LT hours]   :   7  18
IMIN, IMAX of region    :  56  67
JMIN, JMAX of region    :  27  37
LMIN, LMAX of region    :   1  47
```
>Variables contained in the output file:
```
       CATEGORY ILUN  TRCNAME    TRC            UNIT      TAU0 DATE        DIMENSIONS
   1 : IJ-AVG-$   21       NO      1            ppbv 263832.17 2015020600   12  11  47
   2 : IJ-AVG-$   21       O3      2            ppbv 263832.17 2015020600   12  11  47
   3 : IJ-AVG-$   21      SO2     26            ppbv 263832.17 2015020600   12  11  47
   4 : IJ-AVG-$   21      SO4     27            ppbv 263832.17 2015020600   12  11  47
   5 : IJ-AVG-$   21      NH4     31            ppbv 263832.17 2015020600   12  11  47
   6 : IJ-AVG-$   21      NIT     32            ppbv 263832.17 2015020600   12  11  47
   7 : IJ-AVG-$   21     BCPI     34            ppbv 263832.17 2015020600   12  11  47
   8 : IJ-AVG-$   21     OCPI     35            ppbv 263832.17 2015020600   12  11  47
   9 : IJ-AVG-$   21     BCPO     36            ppbv 263832.17 2015020600   12  11  47
  10 : IJ-AVG-$   21     OCPO     37            ppbv 263832.17 2015020600   12  11  47
  11 : IJ-AVG-$   21     DST1     38            ppbv 263832.17 2015020600   12  11  47
  12 : IJ-AVG-$   21     DST2     39            ppbv 263832.17 2015020600   12  11  47
  13 : IJ-AVG-$   21     SALA     42            ppbv 263832.17 2015020600   12  11  47
  14 : IJ-AVG-$   21      NO2     64            ppbv 263832.17 2015020600   12  11  47
  15 : IJ-AVG-$   21     SOAS     74            ppbv 263832.17 2015020600   12  11  47
  16 : TIME-SER   21    ODCOL  19005        unitless 263832.17 2015020600   12  11   1
  17 : TIME-SER   21   AIRDEN  19007       molec/cm3 263832.17 2015020600   12  11  47
``` 
>What could be the possible reason that caused no AOD output in the output file? Thanks!

**Comment by yantosca** [comment]
Thanks for writing.  It could be that the metadata for the AOD output (OD-MAP-$) was inavertently removed from gamap_mod.F when we cleaned that module up. 

I am wondering if the data is still in the bpch file, but is not being read by GAMAP because there are no entries in the tracerinfo.dat/diaginfo.dat for it.

You can try manually adding the following lines to your tracerinfo.dat file:
```
#==============================================================================
# ND21 diagnostic quantities
#==============================================================================
OPTD     Cloud optical depth            0.000E+00  1    14001 1.000E+00 unitless
CLDTOT   3-D cloud frc                  0.000E+00  1    14002 1.000E+00 unitless
                                        0.000E+00  1    14003 1.000E+00 unitless
OPD      Mineral dust opt depth         0.000E+00  1    14004 1.000E+00 unitless
SD       Mineral dust surface area      0.000E+00  1    14005 1.000E+00 cm2/cm3
OPSO4550 Sulfate opt depth (550  nm)    0.000E+00  1    14006 1.000E+00 unitless
HGSO4    Hygr growth of SO4             0.000E+00  1    14007 1.000E+00 unitless
SSO4     Sulfate surface area           0.000E+00  1    14008 1.000E+00 cm2/cm3
OPBC550  Black carbon opt depth (550  n 0.000E+00  1    14009 1.000E+00 unitless
HGBC     Hygr growth of BC              0.000E+00  1    14010 1.000E+00 unitless
SBC      BC surface area                0.000E+00  1    14011 1.000E+00 cm2/cm3
OPOC550  Org carbon opt depth (550  nm) 0.000E+00  1    14012 1.000E+00 unitless
HGOC     Hygr growth of OC              0.000E+00  1    14013 1.000E+00 unitless
SOC      OC surface area                0.000E+00  1    14014 1.000E+00 cm2/cm3
OPSSa550 Seasalt (accum) opt depth (550 0.000E+00  1    14015 1.000E+00 unitless
HGSSa    Hygr growth of seasalt (accum) 0.000E+00  1    14016 1.000E+00 unitless
SSSa     Seasalt (accum) surface area   0.000E+00  1    14017 1.000E+00 cm2/cm3
OPSSc550 Seasalt (coarse) opt depth (55 0.000E+00  1    14018 1.000E+00 unitless
HGSSc    Hygr growth of seasalt (coarse 0.000E+00  1    14019 1.000E+00 unitless
SSSc     Seasalt (coarse) surface area  0.000E+00  1    14020 1.000E+00 cm2/cm3
OPD1550  Dust bin 1 AOD (550  nm)       0.000E+00  1    14021 1.000E+00 unitless
OPD2550  Dust bin 2 AOD (550  nm)       0.000E+00  1    14022 1.000E+00 unitless
OPD3550  Dust bin 3 AOD (550  nm)       0.000E+00  1    14023 1.000E+00 unitless
OPD4550  Dust bin 4 AOD (550  nm)       0.000E+00  1    14024 1.000E+00 unitless
OPD5550  Dust bin 5 AOD (550  nm)       0.000E+00  1    14025 1.000E+00 unitless
OPD6550  Dust bin 6 AOD (550  nm)       0.000E+00  1    14026 1.000E+00 unitless
OPD7550  Dust bin 7 AOD (550  nm)       0.000E+00  1    14027 1.000E+00 unitless
OPSO4    Seasalt (accum) opt depth (    0.000E+00  1    14028 1.000E+00 unitless
OPBC     BC opt depth (     nm)         0.000E+00  1    14029 1.000E+00 unitless
OPOC     OC opt depth (     nm)         0.000E+00  1    14030 1.000E+00 unitless
OPSSa    Seasalt (accum) opt depth (    0.000E+00  1    14031 1.000E+00 unitless
OPSSc    seasalt (coarse) opt depth (   0.000E+00  1    14032 1.000E+00 unitless
OPD1     Dust bin 1 AOD (     nm)       0.000E+00  1    14033 1.000E+00 unitless
OPD2     Dust bin 2 AOD (     nm)       0.000E+00  1    14034 1.000E+00 unitless
OPD3     Dust bin 3 AOD (     nm)       0.000E+00  1    14035 1.000E+00 unitless
OPD4     Dust bin 4 AOD (     nm)       0.000E+00  1    14036 1.000E+00 unitless
OPD5     Dust bin 5 AOD (     nm)       0.000E+00  1    14037 1.000E+00 unitless
OPD6     Dust bin 6 AOD (     nm)       0.000E+00  1    14038 1.000E+00 unitless
OPD7     Dust bin 7 AOD (     nm)       0.000E+00  1    14039 1.000E+00 unitless
OPSO4    Seasalt (accum) opt depth (    0.000E+00  1    14040 1.000E+00 unitless
OPBC     BC opt depth (     nm)         0.000E+00  1    14041 1.000E+00 unitless
OPOC     OC opt depth (     nm)         0.000E+00  1    14042 1.000E+00 unitless
OPSSa    Seasalt (accum) opt depth (    0.000E+00  1    14043 1.000E+00 unitless
OPSSc    seasalt (coarse) opt depth (   0.000E+00  1    14044 1.000E+00 unitless
OPD1     Dust bin 1 AOD (     nm)       0.000E+00  1    14045 1.000E+00 unitless
OPD2     Dust bin 2 AOD (     nm)       0.000E+00  1    14046 1.000E+00 unitless
OPD3     Dust bin 3 AOD (     nm)       0.000E+00  1    14047 1.000E+00 unitless
OPD4     Dust bin 4 AOD (     nm)       0.000E+00  1    14048 1.000E+00 unitless
OPD5     Dust bin 5 AOD (     nm)       0.000E+00  1    14049 1.000E+00 unitless
OPD6     Dust bin 6 AOD (     nm)       0.000E+00  1    14050 1.000E+00 unitless
OPD7     Dust bin 7 AOD (     nm)       0.000E+00  1    14051 1.000E+00 unitless
ODSLA    SLA AOD (600 nm)               0.000E+00  1    14052 1.000E+00 unitless
SASLA    SLA surface area               0.000E+00  1    14053 1.000E+00 cm2/cm3
NDSLA    SLA number density             0.000E+00  1    14054 1.000E+00 num/cm3
ODSPA    PSC type 1a/2 AOD (600 nm)     0.000E+00  1    14055 1.000E+00 unitless
SASPA    PSC type 1a/2 surface area     0.000E+00  1    14056 1.000E+00 cm2/cm3
NDSPA    SPA number density             0.000E+00  1    14057 1.000E+00 num/cm3
```
and these lines to your diaginfo.dat file:
```
   14000 OD-MAP-$                                 Optical Depths  
```
and then restart IDL and gamap and see if it picks up the OD variables.

NOTE: In the `~/IDL/gamap/input_files` folder, there are copies of generic diaginfo.dat and tracerinfo.dat files with all of this metadata... you can always refer to these.

We are preserving the ND51 diagnostics for now but they will be replaced as soon as we can come up with a netCDF equivalent.

**Comment by zsx-GitHub** [comment]
Thanks Bob! Yes, the output are there. I can see them now after adding these lines in the *.dat files!

**Comment by yantosca** [comment]
Shixian wrote:
> Thanks Bob! It works!

I am closing out this issue.

**Comment by msulprizio** [comment]
A fix to restore the information relevant for the timeseries diagnostics has added in commit 11585ed13a3ae290f23b261b33d6128c64b1e943. This fix will be included in 12.9.0.
